gateway-service/src/main/java/org/zowe/apiml/gateway/config/CacheConfig.java

```diff
@@ -40,30 +43,45 @@ public class CacheConfig {
     private static final String EHCACHE_STORAGE_ENV_PARAM_NAME = "ehcache.disk.store.dir";
     private static final String APIML_CACHE_STORAGE_LOCATION_ENV_PARAM_NAME = "apiml.cache.storage.location";
 
+    @Value("${apiml.caching.enabled:true}")
+    private boolean cacheEnabled;
+
     @PostConstruct
     public void afterPropertiesSet() {
-        if (System.getProperty(EHCACHE_STORAGE_ENV_PARAM_NAME) == null) {
-            String location = System.getProperty(APIML_CACHE_STORAGE_LOCATION_ENV_PARAM_NAME);
-            if (location == null) location = System.getProperty("user.dir");
+        if (cacheEnabled) {
+            if (System.getProperty(EHCACHE_STORAGE_ENV_PARAM_NAME) == null) {
+                String location = System.getProperty(APIML_CACHE_STORAGE_LOCATION_ENV_PARAM_NAME);
+                if (location == null) location = System.getProperty("user.dir");
 
-            System.setProperty(EHCACHE_STORAGE_ENV_PARAM_NAME, location);
+                System.setProperty(EHCACHE_STORAGE_ENV_PARAM_NAME, location);
+            }
+        } else {
+            log.warn("Gateway Service is runnning in NoOp Cache mode. Do not use in production.");
```

Comment:
I would add information on how to enable the cache or at least where they should look in the documentation

Reply:
updated